cloudinit/config/schema.py

```diff
@@ -120,18 +149,17 @@ def annotated_cloudconfig_file(cloudconfig, original_content, schema_errors):
         if col is not None:
             msg = 'Line {line} column {col}: {msg}'.format(
                 line=line, col=col, msg=msg)
-        error_footer.append('# E{0}: {1}'.format(error_count, msg))
-        error_count += 1
     lines = original_content.decode().split('\n')
     error_count = 1
-    for line_number, line in enumerate(lines):
-        errors = errors_by_line[line_number + 1]
+    for line_number, line in enumerate(lines, 1):
+        errors = errors_by_line[line_number]
         if errors:
-            error_label = ','.join(
-                ['E{0}'.format(count + error_count)
-                 for count in range(0, len(errors))])
-            error_count += len(errors)
-            annotated_content.append(line + '\t\t# ' + error_label)
+            error_label = []
+            for error in errors:
+                error_label.append('E{0}'.format(error_count))
+                error_footer.append('# E{0}: {1}'.format(error_count, error))
+                error_count += 1
```

Comment:
If we start at error_count = 1, and then we += 1 for each error in the line, aren't we off by one in total error count?

Reply:
The error_count variable is only used to print out annotated labels "E#" in a 1-based manner instead of 0-based. It is not use to count total errors. Probably could rename this error_index.

Reply:
Yes, let's do that.  The variable name confused me.

Reply:
done.